rs/tests/system_tests.bzl

```diff
@@ -344,10 +344,20 @@ def system_test(
         icos_images["ENV_DEPS__GUESTOS_MALICIOUS_DISK_IMG"] = _guestos_malicous + "disk-img.tar.zst"
         icos_images["ENV_DEPS__GUESTOS_MALICIOUS_UPDATE_IMG"] = _guestos_malicous + "update-img.tar.zst"
 
+    # Always add vector runtime and environment variables
+    env["VECTOR_VM_PATH"] = "$(rootpath //rs/tests:vector_with_log_fetcher_image)"
+
+    deps = ["//rs/tests:vector_with_log_fetcher_image"]
+    for dep in runtime_deps:
+        if dep not in UNIVERSAL_VM_RUNTIME_DEPS:
+            deps.append(dep)
+
+    deps = deps + UNIVERSAL_VM_RUNTIME_DEPS
```

Comment:
Moving this logic here is slightly incorrect.

The reason we need to add the `UNIVERSAL_VM_RUNTIME_DEPS` to the runtime_deps of the `name + "_colocate"` target is that that target spins up a UVM to run the test.

However the `UNIVERSAL_VM_RUNTIME_DEPS` are not always necessary in the non-colocated target below. So adding `UNIVERSAL_VM_RUNTIME_DEPS` to the deps unconditionally will add unnecessary dependencies for a lot of tests.

Reply:
But now all of the tests need `UNIVERSAL_VM_RUNTIME_DEPS` because of vector uvm, am I wrong?